lib/timers.js

```diff
@@ -376,66 +406,41 @@ const clearTimeout = exports.clearTimeout = function(timer) {
 };
 
 
-exports.setInterval = function(callback, repeat) {
+exports.setInterval = function(callback, repeat, arg1, arg2, arg3) {
   if (typeof callback !== 'function') {
     throw new TypeError('"callback" argument must be a function');
   }
 
-  repeat *= 1; // coalesce to number or NaN
+  var len = arguments.length;
+  var args;
+  if (len === 3) {
+    args = [arg1];
+  } else if (len === 4) {
+    args = [arg1, arg2];
+  } else if (len > 4) {
+    args = [arg1, arg2, arg3];
+    for (var i = 5; i < arguments.length; i++)
+      // extend array dynamically, makes .apply run much faster in v6.0.0
+      args[i - 2] = arguments[i];
+  }
+
+  return createRepeatTimeout(callback, repeat, args);
+};
 
-  if (!(repeat >= 1 && repeat <= TIMEOUT_MAX)) {
+function createRepeatTimeout(callback, repeat, args) {
+  repeat *= 1; // coalesce to number or NaN
+  if (!(repeat >= 1 && repeat <= TIMEOUT_MAX))
     repeat = 1; // schedule on next tick, follows browser behaviour
-  }
 
-  var timer = new Timeout(repeat);
-  var length = arguments.length;
-  var ontimeout = callback;
-  switch (length) {
-    case 1:
-    case 2:
-      break;
-    case 3:
-      ontimeout = () => callback.call(timer, arguments[2]);
-      break;
-    case 4:
-      ontimeout = () => callback.call(timer, arguments[2], arguments[3]);
-      break;
-    case 5:
-      ontimeout =
-        () => callback.call(timer, arguments[2], arguments[3], arguments[4]);
-      break;
-    default:
-      var args = new Array(length - 2);
-      for (var i = 2; i < length; i += 1)
-        args[i - 2] = arguments[i];
-      ontimeout = () => callback.apply(timer, args);
-      break;
-  }
-  timer._onTimeout = wrapper;
-  timer._repeat = ontimeout;
+  var timer = new Timeout(repeat, callback, args);
+  timer._repeat = repeat;
```

Comment:
Would you rather I leave the variable set to `null` since it wouldn't be used anymore and add a new property that stores the interval value?